modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/schema/IgniteTable.java

```diff
@@ -19,11 +19,41 @@
 
 import java.util.Map;
 import java.util.function.Supplier;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.util.ImmutableIntList;
+import org.apache.ignite.internal.sql.engine.type.IgniteTypeFactory;
 
 /**
  * Table representation as object in SQL schema.
  */
 public interface IgniteTable extends IgniteDataSource {
+    /**
+     * Checks whether it is possible to update a column with a given index.
+     *
+     * @param colIdx Column index.
+     * @return {@code True} if update operation is allowed for a column with a given index.
+     */
+    boolean isUpdateAllowed(int colIdx);
+
+    /**
+     * Returns row type excluding effectively virtual or hidden fields.
+     *
+     * @param factory Type factory.
+     * @return Row type for INSERT operation.
+     */
+    RelDataType insertRowType(IgniteTypeFactory factory);
```

Comment:
these names looks as action for insert something into table. Maybe will be better to rename it, just for example `insertionRowType `

Reply:
or `insertOperationRowType`

Reply:
renamed to `rowTypeForInsert`